smoke_tests/tests/smoke_test_case.py

```diff
@@ -0,0 +1,27 @@
+import re
+from functools import cache
+from unittest import TestCase
+from urllib.parse import urljoin
+
+import requests
+from requests import Response
+
+
+class SmokeTestCase(TestCase):
+    BPM_HOST = None
+    USER_TOKEN = None
+
+    @staticmethod
+    def build_bpm_url(path: str) -> str:
+        assert SmokeTestCase.BPM_HOST, "ERROR - BPMSmokeTests.BPM_HOST not properly set"
+        if re.match(r"^\s*https?://", SmokeTestCase.BPM_HOST):
+            return urljoin(SmokeTestCase.BPM_HOST, path)
+        else:
+            return urljoin(f"https://{SmokeTestCase.BPM_HOST}", path)
+
+    @staticmethod
+    @cache
```

Comment:
Is this caching responses and if so do we want that for a smoke test harness like this?

Reply:
If I understand it correctly: yes.
The idea is that these should never be doing anything complex, such as expecting different responses for the same input. 
On the other hand, it is convenient for separating out the individual checks we're making on responses for sake of clarity and readability (such as testing the status code and sub-systems separately for status endpoint).

Reply:
Let me ask a different way: how does this simplify things? IME caching adds performance at the expense of complexity--are there performance concerns?

Reply:
I'd say remove it then.